src/slider/index.tsx

```diff
@@ -0,0 +1,177 @@
+import { useUniqueId } from '../internal/hooks/use-unique-id';
+import Tooltip from '../internal/components/tooltip';
+import SliderLabels from './slider-labels.js';
+import { getPercent, getStepArray, findLowerAndHigherVals, THUMB_SIZE } from './utils.js';
+import customCssProps from '../internal/generated/custom-css-properties/index.js';
+
+export { SliderProps };
+
+export default function Slider({
+  value,
+  min,
+  max,
+  onChange,
+  step,
+  disabled,
+  invalid,
+  ariaLabel,
+  referenceValues,
+  tickMarks,
+  hideFillLine,
+  valueFormatter,
+  ...rest
+}: SliderProps) {
+  const { __internalRootRef } = useBaseComponent('Slider');
+  const baseProps = getBaseProps(rest);
+
+  const rangeRef = useRef<HTMLDivElement>(null);
+  const handleRef = useRef<HTMLDivElement>(null);
+  const [showPopover, setShowPopover] = React.useState(false);
+  const [isActive, setIsActive] = React.useState(false);
+  const labelsId = useUniqueId('labels');
+  const hasValue = value !== undefined;
+  const formFieldContext = useFormFieldContext(rest);
+
+  const percent = hasValue && getPercent(Math.max(Math.min(value, max), min), min, max);
+
+  // at the min and max values, the thumb is not centered.
+  const tooltipPositionAdjust =
+    hasValue && (value <= min ? THUMB_SIZE - THUMB_SIZE / 2 : value >= max ? THUMB_SIZE + THUMB_SIZE / 2 : THUMB_SIZE);
+
+  useEffect(() => {
+    if (rangeRef.current) {
+      rangeRef.current.style.width = `${percent}%`;
+    }
+  }, [percent]);
+
+  const popoverContent = hasValue && (
+    <Tooltip value={valueFormatter ? valueFormatter(value) : value} trackRef={handleRef} />
+  );
+
+  const getAriaValueText = () => {
+    if (value !== undefined && valueFormatter && valueFormatter(value)) {
+      return valueFormatter(value);
+    }
+
+    if (value && referenceValues && valueFormatter && !valueFormatter(value)) {
+      const valueArray = [min, ...referenceValues, value, max];
+      const prev = valueFormatter(findLowerAndHigherVals(valueArray, value)?.lower);
+      const next = valueFormatter(findLowerAndHigherVals(valueArray, value)?.higher);
+
+      return `${value}, between ${prev} and ${next}`;
+    }
+
+    return undefined;
+  };
+
+  return (
+    <div {...baseProps} ref={__internalRootRef} className={clsx(baseProps.className, styles.root)}>
+      <div className={styles.slider}>
+        <div
+          ref={handleRef}
+          className={clsx(styles['tooltip-thumb'])}
+          style={{
+            insetInlineStart: `calc(${percent}% - ${tooltipPositionAdjust}px)`,
+          }}
+        />
+        {showPopover && popoverContent}
+        <div
+          className={clsx(styles['slider-track'], {
+            [styles.disabled]: disabled,
+          })}
+        />
+        {step && tickMarks && (
+          <div
+            className={clsx(styles.ticks)}
+            style={{
+              [customCssProps.sliderLabelCount]: (max - min) / step,
```

Comment:
CSS can probably handle that but I would round the result anyways. The count value must be an integer logically.